zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/NoteManager.java

```diff
@@ -181,7 +182,9 @@ public void saveNote(Note note, AuthenticationInfo subject) throws IOException {
     } else {
       addOrUpdateNoteNode(new NoteInfo(note));
       noteCache.putNote(note);
-      this.notebookRepo.save(note, subject);
+      synchronized (this) {
```

Comment:
A comment at this point why a `synchronized` has to be used here would be very helpful.

Reply:
The reason makes it `synchronized` is that the `note` object will be empty in the case of concurrent saveNote, which comes from following code:
```note = notebookRepo.get(noteInfo.getId(), noteInfo.getPath(), AuthenticationInfo.ANONYMOUS);```
And since it's just an abstract function, I just add `synchronized` to make it return non-empty object.

Reply:
Please add this as a comment in the code so it is clear to other contributors.